sed/loader/flash/buffer_handler.py

```diff
@@ -79,160 +129,137 @@ def _schema_check(self) -> None:
+        and it pulse resolved (no longer electron resolved). Both are saved as parquet files.
 
         Args:
-            h5_paths (List[Path]): List of paths to H5 files.
-            folder (Path): Path to the folder for buffer files.
-            prefix (str): Prefix for buffer file names.
-            suffix (str): Suffix for buffer file names.
-            force_recreate (bool): Flag to force recreation of buffer files.
+            paths (dict[str, Path]): Dictionary containing the paths to the H5 and buffer files.
         """
-        # Getting the paths of the buffer files, with subfolder as buffer and no extension
-        self.buffer_paths = initialize_paths(
-            filenames=[h5_path.stem for h5_path in h5_paths],
-            folder=folder,
-            subfolder="buffer",
-            prefix=prefix,
-            suffix=suffix,
-            extension="",
-        )
-        # read only the files that do not exist or if force_recreate is True
-        files_to_read = [
-            force_recreate or not parquet_path.exists() for parquet_path in self.buffer_paths
-        ]
 
-        # Get the list of H5 files to read and the corresponding buffer files to create
-        self.missing_h5_files = list(compress(h5_paths, files_to_read))
-        self.save_paths = list(compress(self.buffer_paths, files_to_read))
+        # Create a DataFrameCreator instance and the h5 file
+        df = DataFrameCreator(config_dataframe=self._config, h5_path=paths["raw"]).df
 
-        print(f"Reading files: {len(self.missing_h5_files)} new files of {len(h5_paths)} total.")
+        # forward fill all the non-electron channels
+        df[self.fill_channels] = df[self.fill_channels].ffill()
 
-    def _save_buffer_file(self, h5_path: Path, parquet_path: Path) -> None:
-        """
-        Creates a single buffer file.
-
-        Args:
-            h5_path (Path): Path to the H5 file.
-            parquet_path (Path): Path to the buffer file.
-        """
-
-        # Create a DataFrameCreator instance and the h5 file
-        df = DataFrameCreator(config_dataframe=self._config, h5_path=h5_path).df
+        # Reset the index of the DataFrame and save both the electron and timed dataframes
+        # electron resolved dataframe
+        electron_channels = get_channels(self._config["channels"], "per_electron")
+        dtypes = get_dtypes(self._config["channels"], "all")
+        df.dropna(subset=electron_channels).astype(dtypes).reset_index().to_parquet(
+            paths["electron"],
+        )
 
-        # Reset the index of the DataFrame and save it as a parquet file
-        df.reset_index().to_parquet(parquet_path)
+        # timed dataframe
+        dtypes = get_dtypes(self._config["channels"], ["per_pulse", "per_train"])
+        # drop the electron channels and only take rows with the first electronId
+        df[self.fill_channels].loc[:, :, 0].astype(dtypes).reset_index().to_parquet(paths["timed"])
```

Comment:
I don't understand why all these operations cannot be done lazily on the loaded dataframe from one source.

Reply:
I have tested it again lazily. The lazy operations take about 4x longer compared to this version.
Dask Lazy:
<img width="1068" alt="image" src="https://github.com/OpenCOMPES/sed/assets/10053031/56624895-525c-4008-b716-733c70e942ce">
files are saved to parquet after resetting index straight to parquet and then loaded by dask. The timing is after that.

This PR:
<img width="1079" alt="image" src="https://github.com/OpenCOMPES/sed/assets/10053031/5a36275d-3b8a-4b74-8101-4246211a3361">

A single dataframe I store with all nans also has a filesize of about 5 MB. The electron and timed parquets size about 2.5 MB each, which is the same storage size.

I used run 44824 which has 20 files.

Reply:
The difference between the "old" version, and here is 209 vs. 366 mbyte, while the current version in v1 branch is 500 mbyte. So indeed, this is actually an improvement. Slightly larger compared to the main version, but acceptable I would say (it also contains more information, i.e. the shots without electrons)